### PR TITLE
Fix environment handling from containers.conf

### PR DESCRIPTION
**Pull request by rhatdan**
Signed-off-by: Daniel J Walsh <dwalsh@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
Fixes https://github.com/containers/libpod/issues/5685

File: cmd/podman/shared/create.go [hunk 1]
```diff
@@ -538,18 +538,29 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
...
 
-	// env overrides any previous variables
-	cmdlineEnv := c.StringSlice("env")
-	if len(cmdlineEnv) > 0 {
-		parsedEnv, err := envLib.ParseSlice(cmdlineEnv)
-		if err != nil {
-			return nil, err
+	if c.IsSet("env") {
+		// env overrides any previous variables
+		cmdlineEnv := c.StringSlice("env")
```
Comment (mheon) [inline cmd/podman/shared/create.go:543]: Woah. Is ENV a string slice? That separates automatically on `,` which means we can't have commas in environment variables - that doesn't seem right?

**Comment by mheon** [comment]
LGTM once things go green. I have some questions, but they're unrelated to this PR.

File: cmd/podman/shared/create.go (hunk 1 above)
Comment (rhatdan) [reply cmd/podman/shared/create.go:543]: Perhaps a different issue, it has been this way for a while.

**Comment by rhatdan** [comment]
@giuseppe PTAL

**Comment by giuseppe** [comment]
LGTM

File: cmd/podman/shared/create.go (hunk 1 above)
Comment (larsks) [reply cmd/podman/shared/create.go:543]: @mheon fwiw, commas in environment variables seems to work just fine.

File: cmd/podman/shared/create.go [hunk 2]
```diff
@@ -538,18 +538,29 @@ func ParseCreateOpts(ctx context.Context, c *GenericCLIResults, runtime *libpod.
...
-		env = envLib.Join(env, parsedEnv)
+	}
+
+	labelsList := []string{}
+	if c.IsSet("label") {
+		labelsList = c.StringArray("label")
+	}
+	labelFiles := []string{}
+	if c.IsSet("label-files") {
+		labelFiles = c.StringSlice("label-file")
```
Comment (TomSweeneyRedHat) [inline cmd/podman/shared/create.go:559]: I think line 558 should be singular, "label-file"

**Comment by TomSweeneyRedHat** [comment]
@rhatdan all kinds of test unhappiness.

File: cmd/podman/shared/create.go (hunk 2 above)
Comment (rhatdan) [reply cmd/podman/shared/create.go:559]: Great find.

**Comment by rhatdan** [comment]
Yes for some reason, my new test will not pass, although works when I run this by hand.  Will continue playing with it.

**Pull request merged**